### DISCUSSION TITLE
Using data-variants, but based on screen breakpoints

### DISCUSSION THREAD

**Discussion by SHxKM** [tailwindlabs/tailwindcss#10866] [Help] [answered]
So I'm looking to offer a dense/sparse toggle for my users. The requirement is that they can pick `dense`/`sparse` based on screen size independently.

Here's how I've currently set the classes for my container:

```HTML
<div
     data-xs-cols="1" <!-- adjustable by user -->
     class="
         grid-cols-1 data-[xs-cols='2']:grid-cols-2
         lg:grid-cols-3
         xl:grid-cols-4
     "
>
...
</div>
```

When I set `data-xs-cols="2"`, the grid correctly changes to display 2 columns on mobile instead of 1. But, it also does that for the `lg` and `xl` breakpoints. I'm guessing Tailwind is going by specificity here?

I've tried:

```
xs:data-[xs-cols='2']:grid-cols-2
```

And

```
data-[xs-cols='2']:xs:grid-cols-2
```

But those didn't work.

I really hope I don't have to write "negating" `data-x` modifiers for each breakpoint. Any ideas?

**Comment by SHxKM** [comment, marked as answer]
I could use [Max-width and dynamic breakpoints](https://tailwindcss.com/blog/tailwindcss-v3-2#max-width-and-dynamic-breakpoints), but given @adamwathan's recommendation:

> As a general rule I would still recommend using min-width breakpoints personally, but this feature does unlock one useful workflow benefit which is not having to undo some style at a different breakpoint.

I'm very wary of going into this territory...

**Reply by wongjn** [reply]
I think this would be the most maintainable solution for your use case here, `max-lg:data-[xs-cols='2']:grid-cols-2` but yes, don't overdo it I guess!

**Reply by SHxKM** [reply]
Yep, got myself in trouble with accepting this feature request 🙂
